### ISSUE TITLE
Incorrect video streaming resolution after changing it via json editor

### ISSUE THREAD

**Issue by GetmanetsIrina** [smartdevicelink/sdl_hmi#547]
**Description:**
Incorrect video streaming resolution after changing it via json editor

**Precondition:**
1. HMI and SDL are started
2. Navigation mobile app is registered

**Steps to reproduce:**
1. Mobile app starts video streaming
2. Go to HMI main menu
3. Menu item `Send VIDEO_STREAMING capabilities`
4. Update video capabilities to any preferred resolution values (for example "resolutionWidth": 320, "resolutionHeight": 200)
5. Press `Send notification`
**Note: HMI shows video with the wrong resolution also in case when the mobile app starts stream with the resolution different from the default(800*480)**

**Expected behavior:**
Mobile app starts streaming with the new resolution 320x200
HMI shows streaming with new resolution 320x200
```
Switching video streaming preset to: 320x200
FFW.BasicCommunication.OnSystemCapabilityUpdated
HMI -> SDL: 
BC.OnSystemCapabilityUpdated(systemCapability = {"preferredResolution":{"resolutionWidth":320,"resolutionHeight":200}}, appId)
```
![image](https://user-images.githubusercontent.com/11210973/112624898-df42b080-8e36-11eb-98a4-301629fecb49.png)
![image](https://user-images.githubusercontent.com/11210973/112625984-46149980-8e38-11eb-931a-769dcb98a07b.png)

**Actual result:**
HMI displays video streaming with the wrong resolution
```
FFW.BasicCommunication.OnSystemCapabilityUpdated
HMI -> SDL: 
BC.OnSystemCapabilityUpdated(systemCapability = {"preferredResolution":{"resolutionWidth":320,"resolutionHeight":200}}, appId)
```
![image](https://user-images.githubusercontent.com/11210973/112626199-8e33bc00-8e38-11eb-88ca-7515d6af7db8.png)
*Note:* If the resolution has been changed using drop-down list - streaming resolution is correct 
If the resolution has been changed using `json` editor - streaming resolution isn't correct

**Environment:**
SDL: https://github.com/smartdevicelink/sdl_core/tree/7.1.0-RC1
HMI: https://github.com/smartdevicelink/sdl_hmi/commits/release/5.5.0 (b1d31f8)

**Comment by GetmanetsIrina** [comment]
@jordynmackool, Please notice this issue is related to [Possibility to update video streaming capabilities during ignition cycle](https://github.com/smartdevicelink/sdl_core/issues/3465) feature that should be released in the 7.1 release.
Luxoft team is going to submit a fix for this issue.

**Comment by jordynmackool** [comment]
@GetmanetsIrina please advise when the fix for this PR will be ready for Livio review?

**Comment by AKalinich-Luxoft** [comment]
> @GetmanetsIrina please advise when the fix for this issue will be ready for Livio review?

@jordynmackool fix for this issue is currently in progress and will be ready for Livio review till the end of this week

**Comment by AKalinich-Luxoft** [comment]
@jordynmackool PR with fix is ready for Livio review

**Comment by jordynmackool** [comment]
closed via https://github.com/smartdevicelink/sdl_hmi/pull/552.

**Comment by GetmanetsIrina** [comment]
@jordynmackool, please notice the issue is reopened because the issue is reproduced on the release branch https://github.com/smartdevicelink/sdl_hmi/tree/release/5.5.0 (bbdce98),
but is not reproduced on the fix branch https://github.com/LuxoftSDL/sdl_hmi/tree/fix/fix_resolution_index_after_editor_changes (951cfe2).

**Comment by JackLivio** [comment]
@GetmanetsIrina Testing on latest release branch if I change the preferred resolution via the reproduction steps I see the new resolution being sent?

 Changed height from 380->280
```
HMI - > SDL[7: 43: 11: 139]: {
	"jsonrpc": "2.0",
	"method": "BasicCommunication.OnSystemCapabilityUpdated",
	"params": {
		"appID": 2080409477,
		"systemCapability": {
			"systemCapabilityType": "VIDEO_STREAMING",
			"videoStreamingCapability": {
				"preferredResolution": {
					"resolutionWidth": 800,
					"resolutionHeight": 280
				},
...
```

**Comment by JackLivio** [comment]
I confirmed the video stream actually restarts as well. Are you sure you tested on the updated release branch? Or are you experiencing a new issue?

**Comment by GetmanetsIrina** [comment]
> Are you sure you tested on the updated release branch?
yes

>Or are you experiencing a new issue?
I see the same behaviour as described in issue

@JackLivio, you are right the HMI sends a notification with the correct resolution but sets the wrong resolution on the screen.

I try to update the resolution from JSON editor to e.g. 480x320 and see the next on the release branch
![release_branch_editor](https://user-images.githubusercontent.com/11210973/114054395-817f8100-9898-11eb-9f20-7d45c4cc88e6.png)
![release_branch_main_screen](https://user-images.githubusercontent.com/11210973/114054492-94925100-9898-11eb-8534-4c350fc3767f.png)


on the fix branch i see the different behavior
![fix_branch_editor](https://user-images.githubusercontent.com/11210973/114054715-c60b1c80-9898-11eb-8e14-0686dde9662c.png)
![fix_branch_main_screen](https://user-images.githubusercontent.com/11210973/114054734-ca373a00-9898-11eb-95a0-c737511101f2.png)
I also click on the screen to see the coordinates in red points, so we can see that HMI shows video in the wrong resolution on release branch according to sent coordinates.

**Comment by JackLivio** [comment]
@GetmanetsIrina Ok thank you for the detailed response.

I confirmed that the HMI is not showing my new resolutions in the drop down and the touch events are using the old resolution. My app at least is streaming in the correct new resolution.

It is strange that the fix branch works. Maybe a regression/conflict related to this fix? https://github.com/smartdevicelink/sdl_hmi/pull/537

**Comment by JackLivio** [comment]
I re-tested the  fix branch using commit hash 951cfe20181e56bd7ff8e8256a541f5d6492f4ff

I see the same "issue" happening as described.

**Comment by GetmanetsIrina** [comment]
> Maybe a regression/conflict related to this fix? #537

I think so too

> I see the same "issue" happening as described as happening on release branch.

It's required to set exactly the same resolution as HMI supports (resolution and scale), in case of some custom resolution setting   HMI will show it wrong

**Comment by JackLivio** [comment]
I was sending a resolution not included in the list (100x100). Im not really sure that json editor should be used to switch the video stream to a different preset, which is what I think you are describing?

In my opinion it should be changing the drop down options and only send the capability update if the active option is changed.

**Comment by GetmanetsIrina** [comment]
> I was sending a resolution not included in the list (100x100). This case should not be excluded, this show there are issues still with the fix branch.

agree

> 
> Im not really sure that json editor should be used to switch the video stream to a different preset, which is what I think you are describing?
> 
> In my opinion it should be changing the drop down options and only send the capability update if the active option is changed.

i believe that this is not the big deal for HMI to support this editor

But there is one important thing in this fix - starting video streaming with not default resolution, in case the app starts stream with **not** default resolution(800x380) HMI will start stream with the default one on the release branch 
and with the correct one on the fix branch.

**Comment by JackLivio** [comment]
@GetmanetsIrina I think i know the difference between the fix branch and the release branch. Release branch now deactivates the app if you go to HMI settings, fix branch keeps the app as activated.

When the app is deactivated there is no active app model therefore the changes in the fix branch are not used at all.

This fix is complicated and I am still not understanding the benefit of the JSON editor, especially when the app is deactivated when going to hmi settings. I dont think your test case is possible/useful anymore.

Because of these reasons + approaching the release date, I would like to remove the json editor from this release, and revisit fixing it in the next release.

**Comment by JackLivio** [comment]
> But there is one important thing in this fix - starting video streaming with not default resolution, in case the app starts stream with not default resolution(800x380) HMI will start stream with the default one on the release branch
and with the correct one on the fix branch.

Regarding this point. Is this related to the json editor at all? This sounds like a separate issue.

**Comment by GetmanetsIrina** [comment]
> Regarding this point. Is this related to the json editor at all? This sounds like a separate issue.

@JackLivio this case does not relate to the json editor, but has the same root cause so PR has update for it.

> Because of these reasons + approaching the release date, I would like to remove the json editor from this release, and revisit fixing it in the next release.

Okay, I don't mind

**Comment by JackLivio** [comment]
@GetmanetsIrina Please see this fix to resolve related disussion item:

> But there is one important thing in this fix - starting video streaming with not default resolution, in case the app starts stream with not default resolution(800x380) HMI will start stream with the default one on the release branch
and with the correct one on the fix branch.


https://github.com/smartdevicelink/sdl_hmi/pull/553

**Comment by JackLivio** [comment]
@GetmanetsIrina @AKalinich-Luxoft Let me know if i can close this issue with merge of #553 or if the issue is only partially resolved.

**Comment by AKalinich-Luxoft** [comment]
@JackLivio looks like your PR is only partially resolves the reported issue.
Please take a look at #556 which extends your #553 and makes it applicable for editor as well.

**Comment by GetmanetsIrina** [comment]
@JackLivio , @AKalinich-Luxoft 

> @GetmanetsIrina Please see this fix to resolve related disussion item:
> 
>     But there is one important thing in this fix - starting video streaming with not default resolution, in case the app starts stream with not default resolution(800x380) HMI will start stream with the default one on the release branch
>     and with the correct one on the fix branch.
> 
> #553

The issue has been checked on the release branch (59a1e3b),
the issue above (streaming start with not default resolution) is not reproduced on the specified commit,
but the issue is only partially resolved because we still have the issue with the editor.

The PR  #556 adds the correct processing for the editor (cases when supported resolutions were set)

**Comment by JackLivio** [comment]
Approved and merged #556
